neurons/validators/scraper_validator.py

```diff
@@ -455,11 +457,48 @@ async def query_and_score(self, strategy=QUERY_MINERS.RANDOM):
 
             dataset = QuestionsDataset()
             tools = random.choice(self.tools)
-
+            # Get current time in UTC
+            current_time = datetime.now(pytz.utc)
+            
+            # Track which UIDs already have recent organic results
+            uids_with_recent_results = []
+            recent_organic_results = []
+            recent_organic_rewards = []
+            
+            # Check organic history for each available UID
+            for uid in self.neuron.available_uids:
+                # Get hotkey for this UID
+                neuron = next((n for n in self.neuron.metagraph.neurons if n.uid == uid), None)
+                if not neuron:
+                    continue
+                    
+                hotkey = neuron.hotkey
+                
+                # Check if this hotkey has organic history
+                if hotkey in self.neuron.organic_query_state.organic_history:
+                    for synapse, is_failed, rewards in self.neuron.organic_query_state.organic_history[hotkey]:
+                        # Only consider if tools match
+                        # if set(synapse.tools) == set(tools):
```

Comment:
1. Check if tools are same 
2. If "Twitter Search" tool is in tools then check date filter. Check `date_filter_type` in synapse to be matching enum string and check end_date to be within 1 hour.